src/java.base/share/classes/sun/security/util/DerInputStream.java

```diff
@@ -204,41 +218,35 @@ public int peekByte() throws IOException {
         return data[pos];
     }
 
-    static int getLength(InputStream in) throws IOException {
-        return getLength(in.read(), in);
-    }
-
     /**
      * Get a length from the input stream, allowing for at most 32 bits of
      * encoding to be used.  (Not the same as getting a tagged integer!)
      *
-     * @param lenByte
-     *
      * @return the length or -1 if indefinite length found.
      * @exception IOException on parsing error or unsupported lengths.
      */
-    static int getLength(int lenByte, InputStream in) throws IOException {
-        int value, tmp;
+    static int getLength(InputStream in) throws IOException {
+        int lenByte = in.read();
         if (lenByte == -1) {
             throw new IOException("Short read of DER length");
         }
+        if (lenByte == 0x80) {
+            return -1;
+        }
 
+        int value, tmp;
         String mdName = "DerInputStream.getLength(): ";
         tmp = lenByte;
         if ((tmp & 0x080) == 0x00) { // short form, 1 byte datum
             value = tmp;
         } else {                     // long form or indefinite
```

Comment:
Can't be indefinite here given the new check (line 233-235 above?) Move the indefinite comment upward?